File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/StudentAssessment.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Settings;
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Staff;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class StudentAssessment : Base
    {
        public int StudentId { get; set; }
        public Student? Student { get; set; } = null;
        public int SchoolClassId { get; set; }
        public SchoolClass? SchoolClass { get; set; } = null;
        public int SpecificOutcomeId { get; set; }
        public SpecificOutcome? SpecificOutcome { get; set; } = null;
        public int GradeId { get; set; }
        public Grade? Grade { get; set; } = null;
        public int SessionId { get; set; }
        public Session? Session { get; set; } = null;
        public int AssessmentTypeId { get; set; }
        public AssessmentType? AssessmentType { get; set; } = null;
        public DateTime? AssessmentDate { get; set; }
        public int StaffDetailsId { get; set; }
        public StaffDetails? StaffDetails { get; set; }
        public string? Description { get; set; } = null;
    }
}

File: SchoolWebApp-Client/src/app/cbe/assessments/models/student-assessment.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentAssessment extends ResourceModel<StudentAssessment> {
    public studentId?: number;
    public student?: any;
    public schoolClassId?: number;
    public schoolClass?: any;
    public specificOutcomeId?: number;
    public specificOutcome?: any;
    public subStrandId?: number;
    public subStrand?: any;
    public strandId?: number;
    public strand?: any;
    public gradeId?: number;
    public grade?: any;
    public sessionId?: number;
    public session?: any;
    public assessmentTypeId?: number;
    public assessmentType?: any;
    public assessmentDate?: string;
    public staffDetailsId?: number;
    public staffDetails?: any;
    public description?: string;

    constructor(model?: Partial<StudentAssessment>) {
        super(model);
    }
}
